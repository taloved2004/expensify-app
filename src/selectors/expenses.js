//Get visible expenses
import moment from 'moment';

const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate,note, minAmount, maxAmount } ) =>{
    return expenses.filter((expense)=>{

        const createdAtMoment = moment(expense.createdAt);
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        const startDateMatch = startDate? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch =  endDate? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const noteMatch = note? expense.note.toLowerCase().includes(note.toLowerCase()) : true;
        const minAmountMatch = minAmount !== 0? (minAmount*100)<=expense.amount : true;
        const maxAmountMatch =  maxAmount !== 0? expense.amount<=(maxAmount*100) : true;
        return textMatch && startDateMatch && endDateMatch&&noteMatch&&minAmountMatch&&maxAmountMatch;
    }).sort((a,b)=>{
        if(sortBy==='date'){
            return a.createdAt<b.createdAt? 1: -1;
        }
        if(sortBy==='amount'){
            return a.amount<b.amount? 1 : -1;
        }
    });
};

export default getVisibleExpenses;