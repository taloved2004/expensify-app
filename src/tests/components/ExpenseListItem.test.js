import React from 'react';
import { shallow } from 'enzyme';
import ExepnseListItem  from '../../components/ExpenseListItem';
import expenses from '../fixures/expenses';

test('should render an expense item', ()=>{
    const wrapper = shallow(<ExepnseListItem expense={expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});