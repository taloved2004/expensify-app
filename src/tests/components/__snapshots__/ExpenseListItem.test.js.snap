// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense item 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/undefined"
>
  <div>
    <h3
      className="list-item__title"
    >
      description: 
    </h3>
    <span
      className="list-item__subtitle"
    >
      January 1st, 1970
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $0.00
  </h3>
</Link>
`;
